BlueprintUI/Sources/Environment/Environment.swift

```diff
@@ -0,0 +1,30 @@
+import Foundation
+
+public protocol EnvironmentKey {
+    associatedtype Value
+
+    static var defaultValue: Self.Value { get }
+}
+
+public struct Environment {
+    public static let empty = Environment()
+
+    private init() { }
+
+    private var values: [ObjectIdentifier: Any] = [:]
+
+    public subscript<K>(key: K.Type) -> K.Value where K: EnvironmentKey {
+        get {
+            let objectId = ObjectIdentifier(key)
+
+            if let value = values[objectId] as? K.Value {
```

Comment:
I think should be

```
if let value = values[objectId] {
   return value as! K.Value
}
```

Since this somehow ending up as a the wrong type should be impossible from how this is implemented, so we'd want to fail out.

Reply:
Sure, shouldn't matter either way.

- [ ] force cast K.Value